cf-custom-resources/lib/desired-count-delegation.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+const aws = require('aws-sdk');
+
+// These are used for test purposes only
+let defaultResponseURL;
+
+/**
+ * Upload a CloudFormation response object to S3.
+ *
+ * @param {object} event the Lambda event payload received by the handler function
+ * @param {object} context the Lambda context received by the handler function
+ * @param {string} responseStatus the response status, either 'SUCCESS' or 'FAILED'
+ * @param {string} physicalResourceId CloudFormation physical resource ID
+ * @param {object} [responseData] arbitrary response data object
+ * @param {string} [reason] reason for failure, if any, to convey to the user
+ * @returns {Promise} Promise that is resolved on success, or rejected on connection error or HTTP error response
+ */
+let report = function (event, context, responseStatus, physicalResourceId, responseData, reason) {
+    return new Promise((resolve, reject) => {
+        const https = require('https');
+        const {
+            URL
+        } = require('url');
+
+        var responseBody = JSON.stringify({
+            Status: responseStatus,
+            Reason: reason,
+            PhysicalResourceId: physicalResourceId || context.logStreamName,
+            StackId: event.StackId,
+            RequestId: event.RequestId,
+            LogicalResourceId: event.LogicalResourceId,
+            Data: responseData
+        });
+
+        const parsedUrl = new URL(event.ResponseURL || defaultResponseURL);
+        const options = {
+            hostname: parsedUrl.hostname,
+            port: 443,
+            path: parsedUrl.pathname + parsedUrl.search,
+            method: 'PUT',
+            headers: {
+                'Content-Type': '',
+                'Content-Length': responseBody.length
+            }
+        };
+
+        https.request(options)
+            .on('error', reject)
+            .on('response', res => {
+                res.resume();
+                if (res.statusCode >= 400) {
+                    reject(new Error(`Error ${res.statusCode}: ${res.statusMessage}`));
+                } else {
+                    resolve();
+                }
+            })
+            .end(responseBody, 'utf8');
+    });
+};
+
+/**
+ * Get the current running task number for a specific task definition.
+ *
+ * @param {string} cluster Name of the ECS cluster.
+ * @param {string} family Name of the task definition family.
+
+ * @returns {number} The running task number.
+ */
+const getRunningTaskCount = async function (cluster, family) {
+  var ecs = new aws.ECS();
+
+  var taskNum = 0
+  var nextToken;
+  do {
+    const resp = await ecs.listTasks({
+      cluster: cluster,
+      family: family,
+      nextToken: nextToken
+    }).promise();
+    taskNum += resp.taskArns.length;
+    nextToken = resp.nextToken;
+  } while (nextToken)
+
+  return taskNum;
+};
+
+/**
+ * Correct desired count handler, invoked by Lambda.
+ */
+exports.desiredCountHandler = async function(event, context) {
```

Comment:
nit: I prefer naming all Lambda entry points to be just `handler` so that it's consistent across functions